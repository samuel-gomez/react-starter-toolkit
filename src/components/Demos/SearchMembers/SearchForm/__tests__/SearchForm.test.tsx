import { createRef } from 'react';
import { useForm, useController, ControllerRenderProps } from 'react-hook-form';
import { render } from '@testing-library/react';
import { emptyFunction } from 'shared/helpers';
import SearchForm, { onChangeValue, InputCustom } from '../SearchForm';
import type { FormValues } from '../SearchForm.container';

describe('SearchForm', () => {
  it('Render <SearchForm />', () => {
    const WrapperSearchForm = () => {
      const { control } = useForm<FormValues>();
      const defaultProps = {
        className: 'af-filter-inline',
        hasErrors: false,
        handleSubmit: jest.fn(),
        onSubmit: jest.fn(),
        confirmClassModifier: 'confirm success',
        control,
      };

      return <SearchForm {...defaultProps} />;
    };

    const { asFragment } = render(<WrapperSearchForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('InputCustom', () => {
  const WrapperSearchForm = ({ message }: { message?: string }) => {
    const { formState, control } = useForm<FormValues>();
    const { field } = useController<FormValues>({ name: 'name', control });
    const defaultProps = {
      field,
      formState: !message
        ? formState
        : {
            ...formState,
            errors: {
              name: {
                type: 'required',
                message,
              },
            },
          },
    };

    return <InputCustom {...defaultProps} />;
  };
  it('Render <InputCustom />', () => {
    const { asFragment } = render(<WrapperSearchForm />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render <InputCustom /> with error', () => {
    const { asFragment } = render(<WrapperSearchForm message="champ requis" />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('onChangeValue', () => {
  it('Should called onChange', () => {
    const onChange = jest.fn();
    const field = {
      name: 'name',
      onChange,
      onBlur: emptyFunction,
      value: '',
      ref: createRef().current,
    } as ControllerRenderProps<FormValues, 'name'>;
    onChangeValue(field)({ value: 'test' });
    expect(onChange).toBeCalledWith('test');
  });
});
