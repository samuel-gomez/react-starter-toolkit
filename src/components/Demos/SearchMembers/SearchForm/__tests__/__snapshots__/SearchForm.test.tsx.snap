// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCustom Render <InputCustom /> 1`] = `
<DocumentFragment>
  <div
    class="af-filter-inline__field"
  >
    <div
      class="col-md-12"
    >
      <div
        class="af-form__text"
      >
        <label
          class="af-form__group-label"
          for="name"
        >
          Nom du membre
        </label>
        <input
          aria-label="name"
          autocomplete="search-name"
          class="af-form__input-text"
          id="name"
          name="name"
          placeholder="John Doe"
          type="text"
          value=""
        />
        <small
          class="af-form__help"
        >
          Minimum 3 caractères pour un nom
        </small>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InputCustom Render <InputCustom /> with error 1`] = `
<DocumentFragment>
  <div
    class="af-filter-inline__field"
  >
    <div
      class="col-md-12"
    >
      <div
        class="af-form__text af-form__text--error"
      >
        <label
          class="af-form__group-label"
          for="name"
        >
          Nom du membre
        </label>
        <input
          aria-label="name"
          autocomplete="search-name"
          class="af-form__input-text"
          id="name"
          name="name"
          placeholder="John Doe"
          type="text"
          value=""
        />
      </div>
      <small
        class="af-form__message af-form__message--error"
      >
        <span
          class="glyphicon glyphicon-exclamation-sign"
        />
        <span
          class="af-form__error-text"
        >
          champ requis
        </span>
      </small>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SearchForm Render <SearchForm /> 1`] = `
<DocumentFragment>
  <form
    autocomplete="off"
    class="af-filter-inline"
    id="form-search-members"
    name="form-search-members"
  >
    <div
      class="af-filter-inline__default af-filter-inline__default--search-distributors"
    >
      <div
        class="af-filter-inline__field"
      >
        <div
          class="col-md-12"
        >
          <div
            class="af-form__text"
          >
            <label
              class="af-form__group-label"
              for="name"
            >
              Nom du membre
            </label>
            <input
              aria-label="name"
              autocomplete="search-name"
              class="af-form__input-text"
              id="name"
              name="name"
              placeholder="John Doe"
              type="text"
              value=""
            />
            <small
              class="af-form__help"
            >
              Minimum 3 caractères pour un nom
            </small>
          </div>
        </div>
      </div>
      <button
        class="btn af-btn af-btn--hasiconLeft af-btn--submitconfirm af-btn--success"
        form="form-search-members"
        type="submit"
      >
        <span
          class="af-btn__text"
        >
          Rechercher
        </span>
        <i
          class="glyphicon glyphicon-search"
        />
      </button>
    </div>
  </form>
</DocumentFragment>
`;
